analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/CompoundAssignment.cs

```diff
@@ -51,6 +52,29 @@ private static ProgramState ProcessNumericalCompoundAssignment(ProgramState stat
         }
     }
 
+    private static ProgramState ProcessDelegateCompoundAssignment(ProgramState state, ICompoundAssignmentOperationWrapper assignment)
+    {
+        // When the -= operator is used on a delegate instance (for unsubscribing),
+        // it can leave the invocation list associated with the delegate empty. In that case the delegate instance will become null.
+        // https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/subtraction-operator#delegate-removal
+        // For this reason, we remove the NotNull constraint from the delegate instance.
+        if (assignment.Target.Type.TypeKind == TypeKind.Delegate
+            && assignment.OperatorKind == BinaryOperatorKind.Subtract
+            && state.HasConstraint(assignment.Target, ObjectConstraint.NotNull))
+        {
+            var value = (state[assignment.Target] ?? SymbolicValue.Empty).WithoutConstraint(ObjectConstraint.NotNull);
+            state = state
+                .SetOperationValue(assignment, value)
+                .SetOperationValue(assignment.Target, value);
+            state = state.SetSymbolValue(assignment.Target.TrackedSymbol(state), value);
```

Comment:
Is `Target` always a `TrackedSymbol`? What about indexers?

Reply:
AFAIK the engine doesn't store Null constraints for indexers, so it's not going to get inside the if statement. I've added a UT to cover it.